google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionRunnerImpl.java

```diff
@@ -793,6 +793,7 @@ public SpannerException onError(SpannerException e, boolean withBeginTransaction
         long delay = -1L;
         if (exceptionToThrow instanceof AbortedException) {
           delay = exceptionToThrow.getRetryDelayInMillis();
+          ((AbortedException) exceptionToThrow).setTransactionID(this.transactionId);
```

Comment:
Should this be 
`this.transactionId` or 
`this.transactionId != null ? this.transactionId : this.getPreviousTransactionId()`

Reply:
I guess `this.transactionId != null ? this.transactionId : this.getPreviousTransactionId()` would cause the transaction priority to be preserved also when the begin call of a transaction fails, right? In that case, I think we should do that.

Reply:
Yeah. I have updated it, Thanks.